pandas/core/strings.py

```diff
@@ -323,15 +328,22 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
+    case : boolean, default True (case sensitive)
+        Must be True if `pat` is a compiled regex
```

Comment:
I find this now more confusing, as "must be True for compiled regex" is strange as it in practice not used for a compiled regex